templates/OperationOptionalQueryParamSnippet.stencil

```diff
@@ -0,0 +1,4 @@
+if let {{ param.key }} = {{ param.path }} {
+    {% include "OperationOptionsDecodingSnippet.stencil" %}
+    queryParams.append("{{ param.key }}", {{ param.value }})
```

Comment:
I'm thinking that "value" might be losing its real meaning. As I look at this, I see that really, there should either be a constant or a property path (with or without a dot). Neither of those is captured by value.

```swift
{% if constant  $}
  queryParams.append("{{ param.key }}", "{{ param.constant }}")
{% else %}
  queryParams.append("{{ param.key }}", {{ param.path }})
{% endif %}
```

Reply:
I change the KeyValue Model so that the param.value points to right value whether the value is from constant, option, signature parameter or client property. So we don't need the %if logic in stencil.